flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/Schema.java

```diff
@@ -221,6 +221,15 @@ public Builder fromColumns(List<UnresolvedColumn> unresolvedColumns) {
             return this;
         }
 
+        /** Apply comment to the previous column. */
+        public Builder withComment(@Nullable String comment) {
+            if (columns.size() > 0) {
```

Comment:
Please throw an `IllegalArgumentException` if there is no preceding column defined.